moshi/src/test/java/com/squareup/moshi/MoshiTest.java

```diff
@@ -995,6 +995,39 @@
     }
   }
 
+  @Test public void deferredAdapterRecursesInfinitely() throws Exception {
```

Comment:
Right.
Sorry, I was reasoning through to myself a few days ago what would happen in this case, as someone was talking to me about not being able to use `nextAdapter` in his factory.
This doesn't make sense as a test.